File: backend/src/main/java/pw/checkers/data/enums/MessageType.java
package pw.checkers.data.enums;

import lombok.Getter;

@Getter
public enum MessageType {
    JOIN_QUEUE("joinQueue"),
    LEAVE_QUEUE("leaveQueue"),
    MOVE("move"),
    POSSIBILITIES("possibilities"),
    REMATCH_REQUEST("rematchRequest"),
    ACCEPT_REMATCH("acceptRematch"),
    DECLINE_REMATCH("declineRematch"),
    LEAVE("leave"),
    REJECTION("rejection"),
    ERROR("error"),
    GAME_CREATED("gameCreated"),
    WAITING("waiting"),
    GAME_END("gameEnd"),
    UNKNOWN("unknown");

    private final String value;

    MessageType(String value) {
        this.value = value;
    }

    public static MessageType fromString(String value) {
        for (MessageType mt : values()) {
            if (mt.value.equalsIgnoreCase(value)) {
                return mt;
            }
        }
        return UNKNOWN;
    }
}

File: frontend/composeApp/src/commonMain/kotlin/pw/checkers/game/data/dto/message/MessageType.kt
package pw.checkers.game.data.dto.message

object MessageType {
    object Incoming {
        const val GAME_CREATED = "gameCreated"
        const val MOVE = "move"
        const val POSSIBILITIES = "possibilities"
        const val WAITING = "waiting"
        const val GAME_ENDING = "gameEnd"
        const val REJECTION = "rejection"
        const val REMATCH_REQUEST = "rematchRequest"
    }

    object Outgoing {
        const val JOIN_QUEUE = "joinQueue"
        const val LEAVE_QUEUE = "leaveQueue"
        const val MOVE = "move"
        const val POSSIBILITIES = "possibilities"
        const val REMATCH_REQUEST = "rematchRequest"
        const val ACCEPT_REMATCH = "acceptRematch"
        const val DECLINE_REMATCH = "declineRematch"
        const val LEAVE = "leave"
    }
}
